File: fixtures/scale/dart/large/lib/ciphers/playfair/playfair_digraphs.dart
/// Splits cleaned text into Playfair digraphs, breaking doubles with X.
String splitDigraphs(String text) {
  final folded = text.replaceAll('J', 'I');
  final sb = StringBuffer();
  var i = 0;
  while (i < folded.length) {
    final first = folded[i];
    sb.write(first);
    if (i + 1 < folded.length && folded[i + 1] != first) {
      sb.write(folded[i + 1]);
      i += 2;
    } else {
      sb.write(first == 'X' ? 'Q' : 'X');
      i += 1;
    }
  }
  return sb.toString();
}

File: fixtures/scale/scala/large/src/cryptobench/ciphers/playfair/PlayfairDigraphs.scala
package cryptobench.ciphers.playfair

/** Splits cleaned text into Playfair digraphs, breaking doubles with X. */
object PlayfairDigraphs {
  def split(text: String): String = {
    val folded = text.replace('J', 'I')
    val sb = new StringBuilder
    var i = 0
    while (i < folded.length) {
      val first = folded.charAt(i)
      sb.append(first)
      if (i + 1 < folded.length && folded.charAt(i + 1) != first) {
        sb.append(folded.charAt(i + 1))
        i += 2
      } else {
        sb.append(if (first == 'X') 'Q' else 'X')
        i += 1
      }
    }
    sb.toString
  }
}
